include/swift/ABI/MetadataValues.h

```diff
@@ -14,14 +14,22 @@
 // includes target-independent information which can be usefully shared
 // between them.
 //
+// This header ought not to include any compiler-specific headers (such as
+// those from `swift/AST`, `swift/SIL`, etc.) since doing so may introduce
+// accidental ABI dependencies on compiler internals.
+//
 //===----------------------------------------------------------------------===//
 
 #ifndef SWIFT_ABI_METADATAVALUES_H
 #define SWIFT_ABI_METADATAVALUES_H
 
 #include "swift/ABI/KeyPath.h"
 #include "swift/ABI/ProtocolDispatchStrategy.h"
+
+// FIXME: this include shouldn't be here, but removing it causes symbol
+// mangling mismatches on Windows for some reason?
 #include "swift/AST/Ownership.h"
```

Comment:
@compnerd I don't have a Windows machine to reproduce, but this include seems to be unintentionally load-bearing on the Windows build. Without it, I get a bunch of missing symbol errors in the Concurrency library:

https://ci-external.swift.org/job/swift-PR-windows/21837/consoleText

```
lld-link: error: undefined symbol: void __attribute__((__swiftcall__)) swift::removeStatusRecord(class swift::AsyncTask *, class swift::TaskStatusRecord *, class swift::ActiveTaskStatus &, class llvm::function_ref<void __cdecl(class swift::ActiveTaskStatus, class swift::ActiveTaskStatus &)>)
lld-link: error: undefined symbol: void __attribute__((__swiftcall__)) swift::updateStatusRecord(class swift::AsyncTask *, class swift::TaskStatusRecord *, class llvm::function_ref<void __cdecl(void)>, class swift::ActiveTaskStatus &, class llvm::function_ref<void __cdecl(class swift::ActiveTaskStatus, class swift::ActiveTaskStatus &)>)
lld-link: error: undefined symbol: bool __attribute__((__swiftcall__)) swift::addStatusRecord(class swift::AsyncTask *, class swift::TaskStatusRecord *, class swift::ActiveTaskStatus &, class llvm::function_ref<bool __cdecl(class swift::ActiveTaskStatus, class swift::ActiveTaskStatus &)>)
lld-link: error: undefined symbol: bool __attribute__((__swiftcall__)) swift::addStatusRecordToSelf(class swift::TaskStatusRecord *, class llvm::function_ref<bool __cdecl(class swift::ActiveTaskStatus, class swift::ActiveTaskStatus &)>)
lld-link: error: undefined symbol: void __attribute__((__swiftcall__)) swift::removeStatusRecordFromSelf(class swift::TaskStatusRecord *, class llvm::function_ref<void __cdecl(class swift::ActiveTaskStatus, class swift::ActiveTaskStatus &)>)
```

A common denominator that @rjmccall noticed is that they're referencing `llvm::function_ref`, rather than the `__swift::__runtime::llvm::function_ref` clone that runtime code ought to be using. So I'm wondering whether this is somehow managing to pull in the real LLVM headers instead of the stdlib's copies. I notice that `stdlib/CMakeLists.txt` tries to make this impossible:

```
# Remove llvm-project/llvm/include directory from -I search part when building
# the stdlib. We have our own fork of LLVM includes (Support, ADT) in
# stdlib/include/llvm and we don't want to silently use headers from LLVM.
get_property(stdlib_include_directories DIRECTORY PROPERTY INCLUDE_DIRECTORIES)
list(REMOVE_ITEM stdlib_include_directories "${LLVM_MAIN_INCLUDE_DIR}")
set_property(DIRECTORY PROPERTY INCLUDE_DIRECTORIES ${stdlib_include_directories})
```

but I'm wondering if that expansion doesn't match exactly on Windows for some reason, so the `REMOVE_ITEM` silently does nothing. At some point it might be nice to make the separation clearer here, by using a different non-inline namespace name and include subdirectory, so that the difference between the original `llvm` definitions and their runtime-specific clones is explicit in the code and these subtleties are easier to debug.

Reply:
Hmm, that is odd; but I agree that we should be cleaning this up rather than trying to get away with the incorrect inclusion. (CC: @etcwilde)

Reply:
Oh, yeah, I recall trying to split that out in https://github.com/apple/swift/pull/65133. I don't remember what failed, probably this. I do have a Windows box that I believe can build Swift if you want to poke at it tomorrow, @jckarter. Or I can try to remember to look ¯\_(ツ)_/¯